app/serializers/credentials_serializer.rb

```diff
@@ -3,40 +3,40 @@ def initialize(user: , application:)
     @user, @application = user, application
   end
 
-  def serialize
-    {
-      user: serialized_user,
-      profile: serialized_profile,
-      roles: serialized_roles,
-    }
-  end
-
-  def to_json(opts = {})
-    serialize.to_json opts
+  def as_json(opts = {})
+    serialize
   end
 
   private
 
   attr_reader :user, :application
 
+  def serialize
+    {
+      "user" => serialized_user,
+      "profile" => serialized_profile,
+      "roles" => serialized_roles,
+    }
+  end
+
   def serialized_user
     {
-      email: user.email,
+      "email" => user.email,
     }
   end
 
   def serialized_profile
     {
-      email: profile.email,
-      name:  profile.name,
-      telephone: profile.tel,
-      mobile: profile.mobile,
-      address: {
-        full_address: profile.address,
-        postcode: profile.postcode,
+      "email" => profile.email,
```

Comment:
@pixeltrix Out of interest, where is this contract documented in the Rails docs? The ActiveModel::Errors [as_json() docs](http://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-as_json) show symbol keys returned.

ActiveModel::Serializers doesn't follow this convention. Its `as_json()` tests expect a combination of string and symbol keys:
https://github.com/rails-api/active_model_serializers/blob/3a60633545a1f8dfc81b6e394c49be2d35ab0e06/test/serializers/meta_test.rb

@pedrosmmoreira It appears consumers of the API will always see string keys, as `to_json()` represents hash symbol keys as string keys in the JSON string returned in the HTTP response.